src/plugins/unified_search/public/dataview_picker/change_dataview.tsx

```diff
@@ -309,17 +310,25 @@ export function ChangeDataView({
           onChangeDataView={async (newId) => {
             setSelectedDataViewId(newId);
             setPopoverIsOpen(false);
-            if (isTextBasedLangSelected && !isTextLangTransitionModalDismissed) {
-              setIsTextLangTransitionModalVisible(true);
-            } else if (isTextBasedLangSelected && isTextLangTransitionModalDismissed) {
-              setIsTextBasedLangSelected(false);
-              // clean up the Text based language query
-              onTextLangQuerySubmit?.({
-                language: 'kuery',
-                query: '',
-              });
-              onChangeDataView(newId);
-              setTriggerLabel(trigger.label);
+
+            if (isTextBasedLangSelected) {
+              const showTransitionModal =
+                !isTextLangTransitionModalDismissed &&
+                (!shouldShowTextBasedLanguageTransitionModal ||
```

Comment:
I find this check difficult to read to be honest. Do we really need this to be a function?

Reply:
Yeah, I went back and forth on this a bit myself. I ended up making it a callback since I was worried a boolean might mislead consumers into thinking they can control when the modal is visible, and an optional boolean that defaults to `true` might be more tricky to reason about. But since you find the callback approach less readable anyway, I think going with a boolean makes sense instead. Updated to a boolean here, and just made it default to `true` in the component props: https://github.com/elastic/kibana/pull/177107/commits/6d32f65c2a2af390df64d3e1eff51a0f67d94f9a.